plugins/apache-airflow/src/components/DagTableComponent/DagTableComponent.tsx

```diff
@@ -31,101 +31,151 @@ import Tooltip from '@material-ui/core/Tooltip';
+        <div>
+          <Typography variant="subtitle2" gutterBottom noWrap>
+            {row.id}
+          </Typography>
+          <Box display="flex" alignItems="center">
+            {row.tags?.map((tag, ix) => (
+              <Chip label={tag.name} key={ix} size="small" />
+            ))}
+          </Box>
+        </div>
+      ),
+      width: '50%',
+      disableClick: true,
+      hidden: hiddenColumns.some(field => field === 'id'),
+    },
+    {
+      title: 'Runs',
+      field: 'runs',
+      render: (row: Partial<DagTableRow>) => (
+        <LatestDagRunsStatus dagId={row.dag_id || ''} />
+      ),
+      width: '10%',
+      disableClick: true,
+      hidden: hiddenColumns.some(field => field === 'runs'),
+    },
+    {
+      title: 'Owner',
+      field: 'owners',
+      render: (row: Partial<DagTableRow>) => (
+        <Box display="flex" alignItems="center">
+          {row.owners?.map((owner, ix) => (
+            <Chip label={owner} key={ix} size="small" />
+          ))}
+        </Box>
+      ),
+      width: '10%',
+      disableClick: true,
+      hidden: hiddenColumns.some(field => field === 'owners'),
+    },
+    {
+      title: 'Active',
+      field: 'active',
+      render: (row: Partial<DagTableRow>) => (
+        <Box display="flex" alignItems="center">
+          {row.is_active ? <StatusOK /> : <StatusError />}
+          <Typography variant="body2">
+            {row.is_active ? 'Active' : 'Inactive'}
+          </Typography>
+        </Box>
+      ),
+      width: '10%',
+      disableClick: true,
+      hidden: hiddenColumns.some(field => field === 'active'),
+    },
+    {
+      title: 'Schedule',
+      field: 'schedule',
+      render: (row: Partial<DagTableRow>) => (
+        <ScheduleIntervalLabel interval={row.schedule_interval} />
+      ),
+      width: '10%',
+      disableClick: true,
+      hidden: hiddenColumns.some(field => field === 'schedule'),
+    },
+    {
+      title: 'Link',
+      field: 'dagUrl',
+      render: (row: Partial<DagTableRow>) => (
+        <a href={row.dagUrl}>
+          <IconButton aria-label="details">
+            <OpenInBrowserIcon />
+          </IconButton>
+        </a>
+      ),
+      width: '5%',
+      disableClick: true,
+      hidden: hiddenColumns.some(field => field === 'dagUrl'),
+    },
+  ];
+
   return (
     <Table
       title="DAGs"
-      options={{ pageSize: 5 }}
+      options={{ pageSize: 5, columnsButton: true }}
       columns={columns}
       data={dags}
+      onRowClick={(_event, rowData) => rowClick(rowData)}
+      onChangeColumnHidden={(column, hidden) => {
+        if (column.field) {
+          let newHiddenColumns: string[];
+          if (hidden) {
+            newHiddenColumns = hiddenColumns.concat(column.field);
+          } else {
+            newHiddenColumns = hiddenColumns.filter(v => v !== column.field);
+          }
+          setHiddenColumns(newHiddenColumns);
+          window.localStorage.setItem(
```

Comment:
You could see usages of `storageApiRef` instead of using local storage here in the main repo - if you want this to persist for the current user across sessions and devices